examples/smol_vlm_video/src/video_demo.rs

```diff
@@ -0,0 +1,180 @@
+use kornia::{
+    image::{Image, ImageSize},
+    imgproc::{self, color::YuvToRgbMode},
+    io::{
+        fps_counter::FpsCounter,
+        jpeg,
+        v4l::{camera_control, PixelFormat, V4LCameraConfig, V4lVideoCapture},
+    },
+    tensor::CpuAllocator,
+};
+use kornia_vlm::smolvlm::{utils::SmolVlmConfig, SmolVlm};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+pub fn video_demo(args: &crate::Args) -> Result<(), Box<dyn std::error::Error>> {
+    // ip address of the device with the local rerun viewer you want the data send to
+    let ip_address = "192.168.1.4";
+    let port = 9999;
+
+    let rec = rerun::RecordingStreamBuilder::new("SmolVLM Example: Live Captioning")
+        .connect_grpc_opts(
+            format!("rerun+http://{ip_address}:{port}/proxy"),
+            rerun::default_flush_timeout(),
+        )?;
+
+    // Create the cancellation token for the video capture
+    let cancel_token = Arc::new(AtomicBool::new(false));
+    let cancel_token_clone = Arc::clone(&cancel_token);
+
+    // register a signal handler for graceful shutdown
+    ctrlc::set_handler(move || {
```

Comment:
you could do this instead to create a scoped variable
```suggestion
    ctrlc::set_handler({
      let cancel_token = cancel_token.clone();
      move || {
```